corehq/apps/case_search/utils.py

```diff
@@ -103,3 +104,20 @@ def _add_case_property_queries(self):
                 to_remove = re.escape(removal_regex.regex)
                 value = re.sub(to_remove, '', value)
             self.search_es = self.search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+
+    def _add_daterange_queries(self):
+        # Add query for specially formatted daterange param
+        #   The format is __range__YYYY-MM-DD__YYYY-MM-DD, which is
+        #   used by App manager case-search feature
+        pattern = re.compile(r'__range__\d{4}-\d{2}-\d{2}__\d{4}-\d{2}-\d{2}')
+        drop_keys = []
+        for key, val in self.criteria.items():
+            if val.startswith('__range__'):
+                match = pattern.match(val)
+                if match:
+                    [_, _, startdate, enddate] = val.split('__')
+                    drop_keys.append(key)
+                    self.search_es = self.search_es.date_range_case_property_query(
+                        key, gte=startdate, lte=enddate)
+        for key in drop_keys:
+            self.criteria.pop(key)
```

Comment:
I think this is fine, but this now means that the `_add_case_property_queries` method is reliant on this one always being called first. It might be worth also checking in that method that the value doesn't start with `__range__` to prevent these specially formatted values from getting accidentally passed there (as we do with `UNSEARCHABLE_KEYS`)

Reply:
Sure.